src/repr/scalar/datetime.rs

```diff
@@ -659,21 +665,77 @@ fn fill_pdt_date(
 ) -> Result<()> {
     use TimeStrToken::*;
 
-    let expected = vec![
-        Num(0), // year
-        Dash,
-        Num(0), // month
-        Dash,
-        Num(0), // day
-        Space,
-        DateTimeDelimiter,
+    // Rewrite tokens that are in the format YYYYMMDDD into YYYY-MM-DD.
+    let retokenize_front = if let Some(Num(val)) = actual.front() {
+        // i.e. has 8 digits
+        *val > 9_999_999 && *val < 100_000_000
+    } else {
+        false
+    };
+
+    if retokenize_front {
+        let mut retokenized_front = if let Num(val) = actual.pop_front().unwrap() {
+            let ymd_string = val.to_string();
+            let y = &ymd_string[0..4];
+            let m = &ymd_string[4..6];
+            let d = &ymd_string[6..8];
+            tokenize_time_str(&format!("{}-{}-{}", y, m, d))?
+        } else {
+            VecDeque::new()
+        };
+
+        while !retokenized_front.is_empty() {
```

Comment:
Err, this should be:

```rust
while let Some(t) = retokenized_front.pop_back() {
    actual.push_front(t);
}
```

made this change locally and will push with any other feedback